torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -536,11 +537,22 @@ class TORCH_CUDA_API Allocate : public Expr {
     return buffer_->getDataType().value();
   }
 
+  Allocate* alias() const {
+    return alias_;
+  }
+
+  void setAlias(Allocate* alias) {
+    TORCH_INTERNAL_ASSERT(alias->getMemoryType() == memory_type_);
+    // TODO: TORCH_INTERNAL_ASSERT(alias->size() == size_);
+    alias_ = alias;
+  }
+
  private:
   Val* buffer_ = nullptr;
   MemoryType memory_type_ = MemoryType::Local;
   Val* size_ = nullptr;
   bool zero_init_ = false;
+  Allocate* alias_ = nullptr;
```

Comment:
Can you comment this new member, explaining what it tracks? (... Do we have a linked chain of aliases? Or pointers to a master node? ... )

Reply:
It is a part of a linked chain of aliases.